design-proposals/instancetype.kubevirt.io/support-declarative-vm-management.md

```diff
@@ -0,0 +1,277 @@
+# Overview
+
+At present a VirtualMachine using instance types and/or preferences will have runtime derived data such as the name of a ControllerRevision capturing the state of each resource mutated into the core spec during submission.
+
+This breaks declarative management of VirtualMachines as an owner has no way of pre-populating these runtime derived values and will always see changes made to the spec of their VirtualMachine after submission.
+
+## Motivation
+
+This design proposal aims to enable declarative management of VirtualMachines using instance types and/or preferences by using status to track runtime derived data while retaining all existing behavior and lifecycle support of a VirtualMachine using an instance type and/or preference.
+
+## Goals
+
+* No runtime derived instance type or preference data should be mutated into the core VirtualMachine spec during submission  
+* Users should still be able to snapshot and restore a VirtualMachine using instance types and/or preferences  
+* Users should still be able to switch between instance types potentially invoking vCPU and memory hot plug
+
+## Non Goals
+
+* Deduplication of instance type and preference ControllerRevisions  
+* A new instancetype API group version to v1beta2, all of the API changes actually land in the core v1 API group.
+
+## User Stories
+
+* As a VirtualMachine owner I want to declaratively manage my VirtualMachines using instance types and/or preferences  
+* As a VirtualMachine owner I want existing VirtualMachine lifecycle features to continue to work such as snapshot and restore  
+* As a VirtualMachine owner I want an easy to use mechanism for switching between instance types and preferences
+
+## Repos
+
+* kubevirt/kubevirt
+
+# Design
+
+## `spec.{instancetype,preference}.revisionName` deprecation
+
+The original RevisionName field of the {Instancetype,Preference}Matchers will be deprecated and no longer populated by the VirtualMachine controller.
+
+Any existing or new values will be mirrored into the new `status.{instancetype,preference}` fields discussed below.
```

Comment:
Does it mean it works identical to the new `controllerRevisionRef` field?

Reply:
Yes correct